source/include/dqm4hep/DBInterface.h

```diff
@@ -45,104 +45,181 @@ namespace dqm4hep {
 
-      /** Send query to database.
-       *  The result is streamed in type T using DQM4HEP::stringToType()
+      /** 
+       *  @brief  The result is streamed in type T using DQM4HEP::stringToType()
+       *
+       *  @param  query the mysql query
+       *  @param  result the result to handle in the specified type
        */
       template <typename T>
       StatusCode query(const std::string &query, T &result);
 
-      /** Send query to database.
-       *  The result is streamed in a vector.
-       *  Each value of the vector is streamed using DQM4HEP::stringToType()
+      /** 
+       *  @brief  Send query to database.
+       *          The result is streamed in a vector.
+       *          Each value of the vector is streamed using DQM4HEP::stringToType()
+       *
+       *  @param  query the mysql query
+       *  @param  result the result to handle as a vector of specified type
        */
       template <typename T>
       StatusCode queryVector(const std::string &query, std::vector<T> &result);
 
-      /** Send query to database
-       *  The result is passed in the handler function for user parsing
+      /** 
+       *  @brief  Send query to database
+       *          The result is passed in the handler function for user parsing
+       *
+       *  @param  query the mysql query
+       *  @param  handler a function to handle the mysql query result (lambda or static function)
        */
       template <typename Handler>
       StatusCode queryAndHandle(const std::string &query, Handler handler);
 
-      /** Execute the query (no result expected)
+      /** 
+       *  @brief  Execute the query (no result expected)
+       *
+       *  @param  query the mysql query
        */
       StatusCode execute(const std::string &query);
 
-      /** Whether a connection is being handled
+      /** 
+       *  @brief  Whether a connection is being handled
        */
       bool isConnected();
 
-      // DQM4HEP SPECIFIC
-      // DATABASE INTERFACE
-
-      /** Query a config file content stored in the DQM4HEP data base (table CONFIG_FILES)
+      /**
+       *  @brief  Create a new parameter table
+       *  
+       *  @param  table the table name
+       *  @param  ifNotExists whether to check for table existence on creation
+       *  @param  dropExistingTable whether to drop the existing table before creating
        */
-      StatusCode queryConfigFileContent(const std::string &configFileName, std::string &fileContents);
+      StatusCode createParameterTable(const std::string &table, bool ifNotExists = true, bool dropExistingTable = false);
 
-      /** Query a config file description
+      /**
+       *  @brief  Empty a parameter table (uses the TRUNCATE statement)
+       *  
+       *  @param  table the table name
+       */
+      StatusCode emptyParameterTable(const std::string &table);
+      
+      /**
+       *  @brief  Set the parameters in the table.
+       *          The parameter table must exist.
```

Comment:
What happens if the table doesn't exists?
Might be useful to add a check if the table exists, throw a StatusCode `STATUS_CODE_INVALID_PARAMETER` or `STATUS_CODE_NOT_INITIALIZED` if not.

Would need to add this check in most function.

Reply:
The query will fail if the table doesn't exist. This will happen on the `execute()` call in this function.
So, this is not a problem.